consul/src/main/java/org/secondbase/consul/ConsulModule.java

```diff
@@ -0,0 +1,208 @@
+package org.secondbase.consul;
+
+import com.google.common.net.HostAndPort;
+import com.orbitz.consul.Consul;
+import com.orbitz.consul.model.agent.Registration;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.secondbase.core.SecondBase;
+import org.secondbase.core.SecondBaseException;
+import org.secondbase.core.config.SecondBaseModule;
+
+/**
+ * Consul module for SecondBase. Handles registration of services in consul.
+ */
+public final class ConsulModule implements SecondBaseModule {
+
+    private static final Logger LOG = Logger.getLogger(ConsulModule.class.getName());
+
+    private static final ScheduledExecutorService consulKeepAlive
+            = Executors.newSingleThreadScheduledExecutor();
+    private static final long keepAliveInitialDelaySec = 0L;
+    private static final long keepAlivePeriodSec = 30L;
+    static final DeregisterThread deregisterThread = new DeregisterThread();
+    private static final AtomicBoolean isDeregisterThreadActive = new AtomicBoolean(false);
+
+    private static Consul consulClient;
+
+    @Override
+    public void load(final SecondBase secondBase) {
+        secondBase.getFlags().loadOpts(ConsulModuleConfiguration.class);
+    }
+
+    @Override
+    public void init() throws SecondBaseException {
+        if (!ConsulModuleConfiguration.serviceName.isEmpty()
```

Comment:
This happens too quietly. Whether or not the service is registered one can only see in the consul ui. We need some logging for some or all of the cases. "Registered service x at consul y at port a and health path b" or something like that.